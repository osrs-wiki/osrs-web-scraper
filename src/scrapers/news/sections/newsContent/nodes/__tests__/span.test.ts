import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import spanParser from "../span";

describe("span node", () => {
  test("Span should parse and render", () => {
    const root = parse("<span><b>test</b></span>");
    const builder = new MediaWikiBuilder();
    builder.addContents([spanParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Span with font color should parse and render", () => {
    const root = parse("<span style='color: #ffff00'><b>test</b></span>");
    const builder = new MediaWikiBuilder();
    builder.addContents([spanParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
