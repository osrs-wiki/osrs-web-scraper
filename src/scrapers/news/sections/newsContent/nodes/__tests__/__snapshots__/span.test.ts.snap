// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`span node Span should parse and render 1`] = `"'''test'''"`;

exports[`span node Span with font color should parse and render 1`] = `"<span color=\\"#ffff00\\">'''test'''</span>"`;
